BedrockServer.cpp

```diff
@@ -2148,54 +2148,38 @@ void BedrockServer::handleSocket(Socket* s) {
+                break;
+            } 
+        }
 
-            // A result > 0 indicates a socket is ready to read (or has an error).
-            if (pollResult > 0) {
-                // We can now safely call `recv` without blocking. If it returns true, we've got some data, so we'll
-                // break out of this loop and see if we can do something with that data (like create a command).
+        // If the above loop didn't close the socket due to inactivity at shutdown, let's handle the activity.
+        if (s->state != STCPManager::Socket::CLOSED) {
+            if (pollResult < 0) {
+                // This is an exceptional case, we'll just kill the socket if this happens and let the client reconnect.
+                SINFO("Poll failed: " << strerror(errno));
+                s->state = Socket::CLOSED;
+                ::shutdown(s->s, SHUT_RDWR);
+            } else {
+                // We've either got new data, or an error on the socket. Let's determine which by trying to read.
                 if (!s->recv()) {
-                    // But if it returns false, this means our connection is dead and we should clean it up. We'll drop
-                    // to the bottom of the whole function here.
-                    exitMainLoop = true;
+                    // If reading failed, then the socket was closed.
+                    s->state = Socket::CLOSED;
+                    ::shutdown(s->s, SHUT_RDWR);
                 }
-                // We break out of the inner loop in either case.
-                break;
-            } else if (pollResult == 0) {
-                // Timeout. If we're shutting down, exit main loop, otherwise, try again.
-                if (_shutdownState.load() != RUNNING) {
-                    SINFO("Socket thread exiting because no data and shutting down.");
-                    exitMainLoop = true;
-                    break;
-                }
-            } else {
-                // This is an exception case, we'll just kill the socket if this happens and let the client
-                // reconnect.
-                SINFO("Poll failed: " << strerror(errno));
-                exitMainLoop = true;
-                break;
             }
         }
 
-        // If our poll loop needed to exit from the main loop, we do that now.
-        if (exitMainLoop) {
-            break;
-        }
-
-        // Handle any activity.
-        if (s->state.load() == STCPManager::Socket::CLOSED) {
-            // If the socket got closed, we can jump to the end, there's nothing to do.
-            // FUTURE IMPROVEMENT: What if a client sent us a "fire and forget" command and immediately closed the
-            // socket without waiting for the `202` response. Should we handle that case?
-            break;
-        } else if (s->state.load() == STCPManager::Socket::CONNECTED) {
+        // Now, if the socket hasn't been closed, we'll try to handle the new data on from it appropriately.
```

Comment:
not sure where the typo is in this comment, I think this is what's intended?

```suggestion
        // Now, if the socket hasn't been closed, we'll try to handle the new data from it appropriately.
```